app/src/main/java/io/homeassistant/companion/android/settings/notification/views/NotificationChannelView.kt

```diff
@@ -0,0 +1,115 @@
+            item {
+                Text(
+                    text = stringResource(id = R.string.notification_channels_description),
+                    modifier = Modifier.padding(bottom = 20.dp)
+                )
+                Divider()
+            }
+
+            items(sortedChannelList.size) { index ->
+                val channel = sortedChannelList[index]
+                Row(
+                    modifier = Modifier
+                        .padding(10.dp)
+                        .fillMaxWidth(),
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.SpaceBetween
+                ) {
+                    Text(
+                        text = channel.name.toString().take(30)
+                    )
+                    Row(
+                        modifier = Modifier.padding(start = 12.dp)
+                    ) {
+                        Icon(
+                            Icons.Filled.Edit,
+                            stringResource(id = R.string.edit_channel),
+                            modifier = Modifier
+                                .clickable { notificationViewModel.editChannelDetails(channel.id) }
+                                .padding(12.dp)
+                        )
+                        if (channel.id != SensorWorkerBase.channelId && channel.id != WebsocketManager.CHANNEL_ID) {
+                            Icon(
+                                Icons.Filled.Delete,
+                                stringResource(id = R.string.delete_channel),
+                                modifier = Modifier
+                                    .clickable {
+                                        notificationViewModel.deleteChannel(channel.id)
+                                        notificationViewModel.updateChannelList()
+                                        scope.launch {
+                                            scaffoldState.snackbarHostState
+                                                .showSnackbar(
+                                                    context.getString(R.string.notification_channel_deleted, channel.name),
+                                                    context.getString(R.string.undo)
+                                                )
+                                                .let {
+                                                    if (it == SnackbarResult.ActionPerformed) {
+                                                        notificationViewModel.createChannel(
+                                                            NotificationChannel(
+                                                                channel.id,
+                                                                channel.name,
+                                                                channel.importance
+                                                            )

```

Comment:
Why don't you reuse the original `channel` instead of using the properties to create a new one?

Reply:
I thought I am using the original channel that provides the data for the item?

Reply:
Yes but you're using it to create anew `NotifictionChannel`. As I understand it, you can also replace these 5 lines with `channel` and it still works.

Reply:
ah I see what you mean, thank you :) for some reason I was under the impression I needed to set the properties separtely.